core/src/main/scala/cats/FunctorFilter.scala

```diff
@@ -0,0 +1,62 @@
+package cats
+
+import simulacrum.typeclass
+
+@typeclass trait FunctorFilter[F[_]] extends Functor[F] {
+
+  /**
+   * A combined [[map]] and [[filter]]. Filtering is handled via `Option`
+   * instead of `Boolean` such that the output type `B` can be different than
+   * the input type `A`.
+   *
+   * Example:
+   * {{{
+   * scala> import cats.implicits._
+   * scala> val m: Map[Int, String] = Map(1 -> "one", 3 -> "three")
+   * scala> val l: List[Int] = List(1, 2, 3, 4)
+   * scala> def asString(i: Int): Option[String] = m.get(i)
+   * scala> l.mapFilter(i => m.get(i))
+   * res0: List[String] = List(one, three)
+   * }}}
+   */
+  def mapFilter[A, B](fa: F[A])(f: A => Option[B]): F[B]
```

Comment:
`def mapConcat[A, B](fa: F[A])(f: A => Iterable[B]): F[B]` something like that.

For instance there is no `FunctorMapConcat[Option]` but there is `FunctorMapConcat[Vector]` and in the case of scalding `FunctorMapConcat[TypedPipe]`.